src/pca.js

```diff
@@ -87,22 +87,36 @@ class PCA {
         return new PCA(true, model);
     }
 
+
     /**
      * Project the dataset into the PCA space
      * @param {Matrix} dataset
+     * @param {number} number of components
      * @return {Matrix} dataset projected in the PCA space
      */
-    predict(dataset) {
-        dataset = new Matrix(dataset);
+    predict(dataset, options = {}) {
+        const {
+           nComponents = -1
+        } = options;
 
+        dataset = new Matrix(dataset);
         if (this.center) {
             dataset.subRowVector(this.means);
             if (this.scale) {
                 dataset.divRowVector(this.stdevs);
             }
         }
 
-        return dataset.mmul(this.U);
+        var predictions = dataset.mmul(this.U);
+        if (nComponents === -1) {
+            return predictions;
+        } else {
+            if (nComponents <= predictions.columns) {
```

Comment:
I'd rather check this before doing the multiplication and throw if the parameter is out of range